agera/src/main/java/com/google/android/agera/Result.java

```diff
@@ -161,6 +166,27 @@ public T get() throws FailedResultException {
     throw new FailedResultException(failure);
   }
 
+
+  /**
+   * Returns a list containing the value if it is present, or an empty list.
+   */
+  @NonNull
+  public List<T> asList() {
+    if (value == null) {
```

Comment:
actually I have a slight optimization idea: immediately set `list` to `emptyList()` if Result is constructed with a Throwable. emptyList() is a system-level constant so it doesn't take space (amortized cost); and whether an object field is null or non-null doesn't affect the memory used.